src/redash_stmo/data_sources/query_status/bundle/query_status.js

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import { render } from 'react-dom';
+import { debounce } from 'lodash';
+
+function getQueryStatus($http, query, data_source_id) {
+  return $http.post('api/queries/check', { query, data_source_id });
+}
+
+class QueryStatus extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getQueryStatus = debounce(getQueryStatus, 500);
+    this.state = {
+      valid: false,
+      report: '',
+    };
+  }
+  componentDidMount() {
+    this.componentDidUpdate();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!prevProps || prevProps.queryText !== this.props.queryText || (prevState && prevState.report !== this.state.report)) {
+      const p = this.getQueryStatus(this.props.$http, this.props.queryText, this.props.dataSourceId);
+      if (p) {
+        p.then((response) => { this.setState(response.data); });
+      }
```

Comment:
Do we care about a fail scenario? I think this help with syntax checking so likely not.